File: Java/DirectoryManager.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.File;
import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.FileReader;
import java.io.InputStream;
import java.io.OutputStreamWriter;
import java.io.Writer;
import java.nio.charset.StandardCharsets;
import java.util.List;
 
public class DirectoryManager
{
    // Compile a list of names of all the objects in the specified directory 
    public String[] ContentArray(String path)
    {
        File dir = new File(path);

        if(dir.isDirectory())
        {
            // Return an array of just file names.
            return dir.list();
        }else
        {
            // Return null if path is not a directory.
            return null;
        }		
    }
	
    // A different version of the ContentArray(). This one returns a list of directory
    // paths at whatever recursive depth the user defines.
    public List<String> PathList(String path, List<String> dirList, int recurseDepth)
    {
        File dir = new File(path);
        File[] dirArray = null;

        if(dir.isDirectory())
        {
            dirArray = dir.listFiles();
            for(int i = 0;i < dirArray.length;i++)
            {		
                dirList.add(dirArray[i].toString());
                if(recurseDepth > 0)
                {
                    if(dir.isDirectory())
                    {
                        PathList(dirArray[i].toString(), dirList, recurseDepth - 1);
                    }
                }
            }
        }
	    
        // If path is not a directory, the list will return null. Other wise it will return what was found.
        return dirList;
    }
	
    // Create a folder
    public boolean CreateFolder(String path)
    {
        File dir = new File(path);
        if(dir.isDirectory())
        {
            // If path entered is not a directory return false.
            return false;
        }else
        {
            if(dir.mkdir())
            {
                // If folder creation is successful, return true.
                return true;
            }else
            {
                // If folder creation is not successful, return false.
                return false;
            }
        }
    }

    // Write text to a file
    public boolean WriteText(String path, String content, boolean append)
    {
        try(Writer writer = new BufferedWriter(
            new OutputStreamWriter(
                new FileOutputStream(path, append), StandardCharsets.UTF_8)))
        {
            if(content.trim().equals(""))
            {
                // If there is noting in the input string, don't add a new line to the file.
                writer.write(content);
            }else
            {
                writer.write(content+"\r\n");
            }

            return true;
        }catch(Exception e)
        {
            e.printStackTrace();
            return false;
        }
    }

    // Read a file and return its text
    public String ReadText(String path)
    {
        String contents = null;

        try(BufferedReader br = new BufferedReader(
            new FileReader((path))))
        {
            String line = "";
            while((line = br.readLine()) != null) 
            {
                // Add each line read into the content string
                contents += line;
            }
        }catch(Exception e)
        {
            e.printStackTrace();
        }
	    
        // If exception is thrown, content will return null. Otherwise it will return what was read from the file.
        return contents;
    }

    // Write bytes to a file.
    // Originally I was going to let user pass a string directly into the function, then 
    // convert it. But, passing bytes directly will allow this function to handle files
    // that only deal with bytes.
    public boolean WriteBytes(String path, byte[] byteArray, boolean append)
    {
        try(FileOutputStream fos = new FileOutputStream(path, append)) 
        {
            fos.write(byteArray);
            return true;
        }catch(Exception e)
	{
            e.printStackTrace();
            return false;
	}
    }
	
    // Read bytes from a file
    public byte[] ReadBytes(String path)
    {
        byte[] getBytes = null;
        File file = new File(path); 
        int fileSize = (int) file.length(); 				
        getBytes = new byte[fileSize];

        try(InputStream is = new FileInputStream(file))
        {
            is.read(getBytes);
            // Use the file size to cycle through the byte array, if you want to read each byte.
            /*for(int i = 0;i < fileSize;i++)
            {
                System.out.print((char) getBytes[i]);
            }*/
        }catch(Exception e)
        {
            e.printStackTrace();
        }
        
        // If exception is thrown, return null. Otherwise return the byte array.
        return getBytes;
    }

    // Read a file and look for a specified string
    public boolean SearchText(String path, String find)
    {
        String file = ReadText(path);
        try
        {
            file.contains(find);
            return true;
        }catch(Exception e)
        {
            e.printStackTrace();
            return false;
        }
    }
	
    // Delete a an object in a directory. 
    // I left out a recursive ability on purpose to allow user to decide if they want to do it on their own.
    // This function may be redundant since one can just use .delete(). I just created it because at one point 
    // I thouhgt it might have had a usefulness.
    /*public boolean Delete(String path)
    {
        try
        {
            File file = new File(path);
            file.delete();
            return true;
        }catch(Exception e)
        {
            e.printStackTrace();
            return false;
        }
    }*/
}

File: CSharp/DirectoryManager.cs
class DirMgr
{
    // Create or append text to a file.
    public bool WriteText(string path, string content, bool append)
    {
        try
        { 
            if(append)
            { 
                // Make sure file exists first.
                if(File.Exists(path))
                {
                    // Add content to existing file.
                    using(StreamWriter write = File.AppendText(path))
                    {
                        write.WriteLine(content);
                    }
                }else
                {
                    // Display error if file doesn't exist.
                    Console.WriteLine("File \"{0}\" does not exist. Content cannot be appended.", path);
                    return false;
                }
            }else
            {
                // Overwrite file with new content.
                using(StreamWriter write = File.CreateText(path))
                {
                    write.WriteLine(content);
                }
            }

            return true;
        }catch(Exception e)
        {
            Console.WriteLine(e.Message);
            return false;
        }
    }

    // Read text from an existing file.
    public string ReadText(string path)
    {
        string content = "";

        // Make sure file exists first.
        if(File.Exists(path))
        {
            // Open the file.
            using(StreamReader read = File.OpenText(path))
            {
                string line = "";
                while ((line = read.ReadLine()) != null)
                {
                    content += line;
                }
            }
        }else
        {
            Console.WriteLine("File \"{0}\" does not exist. Content cannot be read.", path);
            return null;
        }

        // The below examples are more simple ways of doing this task.

        // This one takes all the text in a file and puts it into one string.
        //string text = File.ReadAllText(path);

        // This one takes each line in a file and puts them into their own 
        // element in the array.
        //string[] lines = File.ReadAllLines(path);

        return content;
    }

    // Create or append bytes to a file.
    public bool WriteBytes(string path, byte[] content, bool append)
    {
        try
        { 
            if(append)
            {
                if(File.Exists(path))
                { 
                    using(FileStream stream = new FileStream(path, FileMode.Append))
                    { 
                        for(byte i = 0; i < content.Length; i++)
                        {
                            stream.WriteByte(i);
                        }
                        //stream.Write(content, 0, content.Length);
                    }
                }else
                {
                    // Display error if file doesn't exist.
                    Console.WriteLine("File \"{0}\" does not exist. Content bytes cannot be appended.", path);
                    return false;
                }
            }else
            {
                // Overwrite file with new byte content.
                using(FileStream stream = new FileStream(path, FileMode.Create))
                { 
                    for(byte i = 0; i < content.Length; i++)
                    {
                        stream.WriteByte(i);
                    }
                }
            }

            return true;
        }catch(Exception e)
        {
            Console.WriteLine(e.Message);
            return false;
        }
    }

    public byte[] ReadBytes(string path)
    {
        try
        {
            // Read the bytes from the given file.
            byte[] readBytes = File.ReadAllBytes(path);
            // Use loop to process each byte that was read.
            /*foreach(byte bytes in readBytes)
            {
                Console.Write(bytes + " ");
            }*/

            // Honestly, if bytes are to just be returned, this 
            // whole function may be redundant. But, for personal
            // library purposes--and future ideas--I have added 
            // this this function to this class.
            return readBytes;
        }catch(IOException e)
        {
            Console.WriteLine(e.Message);
            return null;
        }
    }
}
